content/en/blog/2020/large-scale-security-policy-performance-tests/index.md

```diff
@@ -0,0 +1,158 @@
+
+This blog post goes over common Security policies use cases and how the number of Security policies or the number of specific rules in a Security policy can effect the overall latency of requests.
+
+## Setup
+
+There are a wide range of Security policies and many more combinations of those policies. We will go over 6 test cases that were determined to be more common/valuable to users.
+
+The following test cases are run in an environment which consists of a fortio client sending requests to a fortio server, except of that of a the baseline which has no Envoy sidecars deployed. The testing tool used to get the following data is the performance benchmarking tool found [here](https://github.com/istio/tools/tree/master/perf/benchmark)
+{{< image width="55%" ratio="45.34%"
+    link="istio_setup.svg"
+    alt="Environment setup"
+    >}}
+
+The policies that are applied are made to not match with any request or only the very last rule, this makes sure that the RBAC filter has to go through all the rules and policies. This forces the system to never short circuit (when the RBAC filter matches before viewing all the policies). Even though this is not necessarily what will happen in your individual system, this gives data of the worst possible performance of each test case.
+
+## Test cases
+
+1. MTLS STRICT vs plaintext.
+
+1. A single authorization policy with a variable number of principal rules as well as a `PeerAuthentication` policy. The principal rule is dependent on the`PeerAuthentication` policy being applied to the system.
+
+1. A single authorization policy with a variable number of `requestPrincipal` rules as well as a `RequestAuthentication` policy. The `requestPrincipal` is dependent on the `RequestAuthentication` policy being applied to the system.
+
+1. A single authorization policy with variable number of paths vs `sourceIP` rules
+
+1. Variable number of authorization policies each consisting of a single paths or `sourceIP` rule
+
+1. A single `RequestAuthentication` policy with variable number of `JWTRules` rules
+
+## Data
+
+The y axis of each test is the latency in milliseconds, and the x axis is the number of concurrent connections. The x axis of each graph will consist of 3 data points. This first being a small load (qps=100, conn=8), the second a medium load (qps=500, conn=32), and the third being a large load (qps=1000, conn=64)
+
+{{< tabset category-name="platform" >}}
+
+{{< tab name="MTLS vs plainText" category-value="one" >}}
+{{< image width="90%" ratio="45.34%"
+    link="mtls_plaintext.svg"
+    alt="MTLS vs plaintext"
+    caption=""
+    >}}
+The difference between MTLS mode STRICT vs that of plaintext is very small in lower loads and as the number of qps and conn increase the latency of requests with MTLS STRICT rise higher. The additional latency increased in larger loads is minimal compared to that of the increase from having no sidecars vs having sidecars in the plaintext.
```

Comment:
```suggestion
The difference of latency between MTLS mode STRICT and plaintext is very small in lower loads. As the `qps` and `conn` increase, the latency of requests with MTLS STRICT increases. The additional latency increased in larger loads is minimal compared to that of the increase from having no sidecars to having sidecars in the plaintext.
```

Reply:
Following comments in this version have been updated in the latest commit